### DISCUSSION TITLE
繪制組件的preRenderDatas能加入其他的屬性?

### DISCUSSION THREAD

**Discussion by stan88** [zouyaoji/vue-cesium#364] [Q&A]
您好,
請問繪制組件透過preRenderDatas所繪制的圖形
能像下圖這樣給予不同的名稱或VCID?
目前能把先前繪制的圖形VCID存下,
但無法透過JSON格式重繪
謝謝

![2023-02-23_223908](https://user-images.githubusercontent.com/5283322/220939866-b5838620-15c6-4611-a1c8-0f3978a413a4.png)

**Comment by zouyaoji** [comment]
preRenderDatas 不可以的，可以试试下面的办法。拿面举例，默认索引为3
（1）绘制完成后，通过ref模板引用拿到绘制的对象，保存。
const datas = vm.$refs.drawingsRef.getDrawingActionInstances()[3].cmpRef.value.renderDatas.value
（2）重绘时再插入
vm.$refs.drawingsRef.getDrawingActionInstances()[3].cmpRef.value.renderDatas.value.push(...datas)

**Reply by ZhaoSunrise-Lee** [reply]
官方绘制组件里面断点打印，value是null
<img width="865" height="528" alt="a" src="https://github.com/user-attachments/assets/454e712f-ef4d-4335-bfa4-9c8e9437725c" />

**Comment by stan88** [comment]
您好,
您的意思是繪制對象裡的屬性值都必須保存下來?
如果只有PUSH"對象類別","座標","顏色","標識字"等,
是不行的?
謝謝

**Reply by zouyaoji** [reply]
也是可以的呀。都是动态的。
